nighthawk/include/nighthawk/client/benchmark_client.h

```diff
@@ -0,0 +1,73 @@
+#pragma once
+
+#include <functional>
+#include <memory>
+
+#include "envoy/runtime/runtime.h"
+
+#include "nighthawk/common/statistic.h"
+
+namespace Nighthawk {
+namespace Client {
+
+typedef std::function<bool(const std::string, const uint64_t)> CounterFilter;
+
+class BenchmarkClient {
+public:
+  virtual ~BenchmarkClient() = default;
+
+  /**
+   * Initialize will be called on the worker thread after it has started.
+   * @param runtime to be used during initialization.
+   */
+  virtual bool initialize(Envoy::Runtime::Loader& runtime) PURE;
+
+  /**
+   * Terminate will be called on the worker thread before it ends.
+   */
+  virtual void terminate() PURE;
+
+  /**
+   * Turns latency measurement on or off.
+   *
+   * @param measure_latencies true iff latencies should be measured.
+   */
+  virtual void setMeasureLatencies(bool measure_latencies) PURE;
```

Comment:
I wonder if this is exposing an internal implementation detail; why not always measure latencies?

Reply:
This gets used later, we currently do not measure during the warmup phase, and enable latency when we transition out the it

Reply:
Ack, ignore this comment then. Thanks for the explanation.